src/EventStore.Client/EventStoreClientBase.cs

```diff
@@ -96,12 +106,119 @@ public abstract class EventStoreClientBase :
+			// 	}.Uri,
+			// };
+		}
+
+		/// <summary>
+		///	Returns the result of an HTTP Get request based on the client settings.
+		/// </summary>
+		/// <param name="path"></param>
+		/// <param name="userCredentials"></param>
+		/// <param name="onNotFound"></param>
+		/// <param name="cancellationToken"></param>
+		protected async Task<T> HttpGet<T>(string path, UserCredentials? userCredentials, Action onNotFound,
+			CancellationToken cancellationToken) {
+			
+			var request = CreateRequest(path, HttpMethod.Get, userCredentials);
+			var httpResult = await HttpSend(request, onNotFound, cancellationToken).ConfigureAwait(false);
+			var json = await httpResult.Content.ReadAsStringAsync().ConfigureAwait(false);
+			var result = JsonConvert.DeserializeObject<T>(json, _jsonSettings);
+			if (result == null) {
+				throw new InvalidOperationException("Unable to deserialize response into object of type " + typeof(T));
+			}
+			
+			return result;
+		}
+		
+		/// <summary>
+		///	Executes an HTTP Post request based on the client settings.
+		/// </summary>
+		/// <param name="path"></param>
+		/// <param name="userCredentials"></param>
+		/// <param name="onNotFound"></param>
+		/// <param name="cancellationToken"></param>
+		/// <returns></returns>
+		protected async Task HttpPost(string path, UserCredentials? userCredentials, Action onNotFound,
+			CancellationToken cancellationToken) {
+			
+			var request = CreateRequest(path, HttpMethod.Post, userCredentials);
+			await HttpSend(request, onNotFound, cancellationToken).ConfigureAwait(false);
+		}
+		
+		private async Task<HttpResponseMessage> HttpSend(HttpRequestMessage request, Action onNotFound, CancellationToken cancellationToken) {
+			var httpResult = await _httpClient.Value.SendAsync(request, cancellationToken).ConfigureAwait(false);
+			if (httpResult.IsSuccessStatusCode) {
+				return httpResult;
+			}
+
+			if (httpResult.StatusCode == HttpStatusCode.Unauthorized)
+			{
+				throw new AccessDeniedException();
+			}
+			if (httpResult.StatusCode == HttpStatusCode.NotFound) {
+				onNotFound();
+			}
+			
+			throw new Exception($"The HTTP call failed with status code: {httpResult.StatusCode}");
+		}
+
+
+		private HttpRequestMessage CreateRequest(string path, HttpMethod method, UserCredentials? credentials) {
+			credentials ??= Settings.DefaultCredentials;
+			
+			var uri = MakeUrl(path);
+			var httpRequest = new HttpRequestMessage(method, uri);
+			if (credentials != null) {
+				httpRequest.Headers.Add("authorization", $"Basic {GetAuthorizationHeader(credentials)}");
+			}
+
+			httpRequest.Headers.Add("accept", "application/json");
+			
+			return httpRequest;
+		}
+		
+		private static string GetAuthorizationHeader(UserCredentials credentials)
+			=> Convert.ToBase64String(Encoding.ASCII.GetBytes($"{credentials.Username}:{credentials.Password}"));
+
+		private Uri MakeUrl(string path) {
+			var httpEndPoint = Settings.ConnectivitySettings.Address;
```

Comment:
not certain but is this limited to dns discovery?

Reply:
Code is refactored and this part is now coming from the `ChannelInfo`